wordpress-mobile/WordPress-Android
# [Dependency Updates] Update `navigationVersion` to 2.5.3

Parent #17935

This PR updates `navigationVersion` to [2.5.3](https://developer.android.com/jetpack/androidx/releases/navigation#version_242_3).

-----

PS: @ovitrif I added you as the main reviewer, randomly so, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

Build Refactor List:

1. [Rename nav component to navigation version](https://github.com/wordpress-mobile/WordPress-Android/commit/f9727f4d2ab40e7a7ceb74c6de06d31304b8bb3c)
2. [Move navigation below the agp version and plugin ids](https://github.com/wordpress-mobile/WordPress-Android/commit/adb6ea964687a8180cdbc40978238fd641b90928)

Plugin Dependency Refactor List:

1. [Extract google services plugin related version](https://github.com/wordpress-mobile/WordPress-Android/commit/661c5c40da148420631178c552a130eea63d4a12)
2. [Extract sentry plugin related version](https://github.com/wordpress-mobile/WordPress-Android/commit/9e4e63539b9a71f6b64c14b70f5775eecc78a76e)
3. [Extract violation comments plugin related version](https://github.com/wordpress-mobile/WordPress-Android/commit/a5aa51cd26460b446e87ceacde63bdbd75835c70)

-----

## To test:

1. See the dependency tree diff result and verify correctness.
2. Thoroughly smoke test any navigation related functionality both, the WordPress and Jetpack apps, and see if everything is working as expected.
3. In addition to the above smoke test, you can expand the below and follow the inner and more explicitly test steps within:

<details>
    <summary>Image Editing Flow [libs:image-editor]</summary>

ℹ️ This test applies to both, the `WordPress` and `Jetpack` apps.

- Add a new `blog` post.
- Add a new `image` block.
- Choose an image and wait for it to be uploaded within the `image` block.
- Click on the `media options` of this image (top right) and then click `edit`.
- Verify that the `Edit Image` screen is shown and functioning as expected.
- Crop the image and click the `done` menu option (top right).
- Make sure the image is updated accordingly.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour on any or all navigation related functionality, especially the `Image Editing` screen, which is closely related to `androidx.navigation`.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

4. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:


-----

UI Changes testing checklist:
